087-external-certificate-manager.md

```diff
@@ -0,0 +1,294 @@
+
+During a reconciliation Strimzi will check the hash of the certificate stored in `clusterCa.publicCert` Secret to see if an update is needed.
+If the certificate has changed Strimzi will rename the existing cert in the `<CLUSTER_NAME>-cluster-ca-cert` Secret to `ca-YYYY-MM-DDTHH-MM-SSZ.crt` and copy over the new certificate.
+It will also update the `strimzi.io/ca-cert-hash` and increment the `strimzi.io/ca-key-generation` and `strimzi.io/ca-cert-generation` annotations.
+
+#### Issuing end-entity certificates
+
+When Strimzi needs to issue a certificate, instead of using the existing internal mechanism it will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANS in the `Certificate` resource for the end-entity certificate.
+Strimzi will set the `secretName` field in the `Certificate` resource as `<CERT_SECRET>-cm`, where `<CERT_SECRET>` is the name of the Secret Strimzi currently uses, for example `<CLUSTER_NAME>-cluster-operator-certs-cm`.
+Strimzi will wait for the usual operation timeout during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
+When issuing cluster certificates (e.g for each Kafka pod etc), once the certificate has been issued, Strimzi will copy the certificate across from the cert-manager provided Secret into its own existing Secret.
+Strimzi will annotate the Secret it manages with:
+* `strimzi.io/server-cert-hash` annotation with the value being the hash of the certificate in the cert-manager Secret.
+* `strimzi.io/cluster-ca-cert-generation` annotation with the value matching the current value of `strimzi.io/ca-cert-generation` on the `<CLUSTER_NAME>-cluster-ca-cert` Secret.
+
+Similar to today, Strimzi will also add both these annotations to the pod mounting the Secret.
+
+#### Tracking changes to cluster end-entity certificates
+
+Cert-manager will be responsible for renewing all end-entity certificates.
+When a certificate is renewed cert-manager will update the related Secret.
+During a reconciliation Strimzi will check the hash of the certificate stored in cert-manager Secrets to see if an update is needed.
+
+When cert-manager renews an end-entity certificate Strimzi will copy over the new certificate into its own Secret.
+Strimzi will update the `strimzi.io/server-cert-hash` annotation to match the new certificate.
+Strimzi will update the `strimzi.io/cluster-ca-cert-generation` annotation to match the `strimzi.io/ca-cert-generation` annotation on the `<CLUSTER_NAME>-cluster-ca-cert` Secret.
+
+Strimzi will review the `Certificate` resource every time it does a reconciliation to see if any changes to the requested certificate are needed, for example updating SANs.
+
+> ![Issuing new component end-entity certificates](./images/087-new-ee-certs.png)
+>
+> Fig 1: Proposed workflow when cert-manager issues new component end-entity certificates
+
+#### Handling CA key replacements
```

Comment:
Is there any difference between CA key replacement here and simple CA public key renewal? Given the later seems not to be covered in any other section, I assume all public key renewals are handled in the same way as key replacements (as we do with custom CAs today)? Assuming that is the case, it would be worth clarifying it here.